packages/components/menu/src/menu.ts

```diff
@@ -222,65 +217,7 @@ export default defineComponent({
         }
       }
     }
-
-    const flattedChildren = children => {
-      const temp = Array.isArray(children) ? children : [children]
-      const res = []
-      temp.forEach(child => {
-        if (Array.isArray(child.children)) {
-          res.push(...flattedChildren(child.children))
-        } else {
-          res.push(child)
-        }
-      })
-      return res
-    }
-
-    const updateFilteredSlot = async () => {
-      filteredSlot.value = slots.default?.()
-      await nextTick()
-      if (props.mode === 'horizontal') {
-        const items = Array.from(menu.value.childNodes).filter((item: HTMLElement) => item.nodeName !== '#text' || item.nodeValue) as [HTMLElement]
-        const originalSlot = flattedChildren(slots.default?.()) || []
-        if (items.length === originalSlot.length) {
-          const moreItemWidth = 64
-          const paddingLeft = parseInt(getComputedStyle(menu.value).paddingLeft)
-          const paddingRight = parseInt(getComputedStyle(menu.value).paddingRight)
-          const menuWidth = menu.value.clientWidth - paddingLeft - paddingRight
-          let calcWidth = 0
-          let sliceIndex = 0
-          items.forEach((item, index) => {
-            calcWidth += item.offsetWidth || 0
-            if (calcWidth <= menuWidth - moreItemWidth) {
-              sliceIndex = index + 1
-            }
-          })
-          const defaultSlot = originalSlot.slice(0, sliceIndex)
-          const moreSlot = originalSlot.slice(sliceIndex)
-          if (moreSlot?.length) {
-            filteredSlot.value = [
-              ...defaultSlot,
-              h(ElSubMenu, {
-                index: 'sub-menu-more',
-                class: 'el-sub-menu__hide-arrow',
-              }, {
-                title: () => h('i', { class: ['el-icon-more', 'el-sub-menu__icon-more'] }),
-                default: () => moreSlot,
-              }),
-            ]
-          }
-        }
-      }
-    }
-
-    const handleResize = () => {
-      updateFilteredSlot()
-    }
-
-    // watch
-    watch(() => slots.default?.(), () => {
-      updateFilteredSlot()
-    })
+    const handleResize = debounce(() => instance.proxy.$forceUpdate(), 50)
```

Comment:
Why not 
```ts
    const handleResize = debounce(instance.proxy.$forceUpdate, 50)
```

Reply:
This is to call it explicitly, or to prevent the `handleResize` parameter from being passed to `$forceUpdate`